src/server_manager/web_app/ui_components/outline-gcp-create-server-app.ts

```diff
@@ -0,0 +1,342 @@
+  }
+
+  private async handleBillingVerificationNextTap(): Promise<void> {
+    this.billingAccounts = await this.account.listBillingAccounts();
+    if (!this.billingAccounts || this.billingAccounts.length === 0) {
+      // TODO: Show error
+    } else {
+      this.showProjectSetup();
+    }
+  }
+
+  private isProjectSetupNextEnabled(projectId: string, billingAccountId: string): boolean {
+    // TODO: Proper validation
+    return projectId !== '' && billingAccountId !== '';
+  }
+
+  private async handleProjectSetupNextTap(): Promise<void> {
+    this.project =
+        await this.account.createProject(this.selectedProjectId, this.selectedBillingAccountId);
+    this.showRegionPicker();
+  }
+
+  private showBillingAccountSetup(): void {
+    this.currentPage = 'billingAccountSetup';
+  }
+
+  private showProjectSetup(): void {
+    this.currentPage = 'projectSetup';
+  }
+
+  private async showRegionPicker(): Promise<void> {
+    this.currentPage = 'regionPicker';
+    const regionMap = await this.account.listLocations(this.project.id);
+    const locations = Object.entries(regionMap).map(([regionId, zoneIds]) => {
+      return this.createLocationModel(regionId, zoneIds);
+    });
+    this.regionPicker.locations = locations;
+  }
+
+  async start(account: GcpAccount): Promise<void> {
+    this.init();
+    this.account = account;
+
+    const projects = await this.account.listProjects();
+    if (projects && projects.length > 0) {
+      this.project = projects[0];
+      this.showRegionPicker();
+    } else {
+      this.billingAccounts = await this.account.listBillingAccounts();
+      if (!this.billingAccounts || this.billingAccounts.length === 0) {
+        this.showBillingAccountSetup();
+      } else {
+        this.showProjectSetup();
+      }
+    }
+  }
+
+  private init() {
+    this.currentPage = '';
+    this.selectedProjectId = '';
+    this.selectedBillingAccountId = '';
+    this.regionPicker = this.shadowRoot.querySelector('#regionPicker') as OutlineRegionPicker;
+    this.regionPicker.reset();
+  }
+
+  private onProjectIdChanged(event: CustomEvent) {
+    this.selectedProjectId = event.detail.value;
+  }
+  private onBillingAccountSelected(event: CustomEvent) {
+    this.selectedBillingAccountId = event.detail.value;
+  }
+
+  private async onRegionSelected(event: CustomEvent) {
+    event.stopPropagation();
+
+    this.regionPicker.isServerBeingCreated = true;
+    // TODO: Make unique server name
+    const server =
+        await this.account.createServer(this.project.id, 'outline', event.detail.selectedRegionId);
+    const params = {bubbles: true, composed: true, detail: {server}};
+    const serverCreatedEvent = new CustomEvent('gcp-server-created', params);
+    this.dispatchEvent(serverCreatedEvent);
+  }
+
+  private createLocationModel(regionId: string, zoneIds: string[]): Location {
+    return {
+      id: zoneIds.length > 0 ? zoneIds[0] : null,
+      name: LOCATION_MAP.get(regionId),
```

Comment:
Let's have a fallback:
```suggestion
      name: LOCATION_MAP.get(regionId) ?? regionId,
```